### ISSUE TITLE
[Phase 2.5] 損失関数の実装

### ISSUE THREAD

**Issue by Showhey798** [Showhey798/gradflow#11]
## 概要

機械学習の損失関数とその勾配を実装します。

## 実装項目

- [ ] MSELossOperation
- [ ] CrossEntropyLossOperation
- [ ] BCELossOperation (Binary Cross Entropy)
- [ ] NLLLossOperation (Negative Log Likelihood)

## ファイル

- `include/gradflow/autograd/ops/loss.hpp`
- `tests/test_loss_grad.cpp`

## テスト項目

- [ ] LossGradTest::MSELossGradient
- [ ] LossGradTest::CrossEntropyGradient

## 完了基準

- [ ] 数値勾配チェックがすべてパス
- [ ] 損失関数が収束することを簡単な例で確認

## 関連ドキュメント

- [ROADMAP.md Phase 2.5](docs/ROADMAP.md)

## 期間

Week 5

## 依存タスク

- Depends on #10
